pkg/cmd/roachprod/install/expander.go

```diff
@@ -100,26 +114,32 @@ func (e *expander) maybeExpandUIPort(c *SyncedCluster, s string) (string, bool)
 		}
 	}
 
-	return e.maybeExpandMap(c, e.uiPorts, m[1])
+	s, err := e.maybeExpandMap(c, e.uiPorts, m[1])
+	return s, err == nil, err
 }
 
-func (e *expander) maybeExpandStoreDir(c *SyncedCluster, s string) (string, bool) {
+func (e *expander) maybeExpandStoreDir(c *SyncedCluster, s string) (string, bool, error) {
 	if !storeDirRe.MatchString(s) {
-		return s, false
+		return s, false, nil
 	}
-	return c.Impl.NodeDir(c, e.node), true
+	return c.Impl.NodeDir(c, e.node), true, nil
 }
 
-func (e *expander) maybeExpandLogDir(c *SyncedCluster, s string) (string, bool) {
+func (e *expander) maybeExpandLogDir(c *SyncedCluster, s string) (string, bool, error) {
 	if !logDirRe.MatchString(s) {
-		return s, false
+		return s, false, nil
 	}
-	return c.Impl.LogDir(c, e.node), true
+	return c.Impl.LogDir(c, e.node), true, nil
 }
 
-func (e *expander) expand(c *SyncedCluster, arg string) string {
-	return parameterRe.ReplaceAllStringFunc(arg, func(s string) string {
-		type expanderFunc func(*SyncedCluster, string) (string, bool)
+type expanderFunc func(*SyncedCluster, string) (string, bool, error)
```

Comment:
actually this is a good place for the comment (with each expanderFunc linking to this def)